### ISSUE TITLE
EXC_BAD_ACCESS causing application crash

### ISSUE THREAD

**Issue by emiller** [star-micronics/react-native-star-io10#25]
Hello,

We're starting to see hard crashes in our application that appear to be originating from the `StarLogger`, resulting in random crashes in our application. Any insight would be great. Here is information from our Sentry integration:

![image](https://user-images.githubusercontent.com/568053/132412222-01b989ef-8cc2-46de-a767-edc826762e51.png)

Is it possible to disable `StarIO10`'s logger entirely to avoid this issue?

**Comment by emiller** [comment]
Hi @gare-bear -- I would be glad to but we aren't logging anything via the star library, in fact we weren't even aware there was such a facility available to us in `StarIO10`. One thing worth noting is that the exceptions tend to pop up randomly, even when we aren't executing print logic.

Here is a snippet of the print flow though if it helps:

```tsx
  const settings = new StarConnectionSettings();
  settings.interfaceType = interfaceType;
  settings.identifier = identifier;
  settings.autoSwitchInterface = true;

  const printer = new StarPrinter(settings);
  printer.openTimeout = CONNECT_TIMEOUT;

  try {
    await printer.open();
    const status = await printer.getStatus();

    if (status.hasError) {
      if (status.coverOpen) {
        throw new StandardPrintError('cover_open');
      }

      if (status.paperEmpty) {
        throw new StandardPrintError('no_paper');
      }
    }

    var imgs = "<business logic>"

    const builder = this.printCommandBuilder(width, imgs);
    const commands = await builder.getCommands();

    await printer.print(commands);

    if (status.paperNearEmpty) {
      const warning = new StandardPrintWarning('low_paper');
      return new StandardPrintResult('ok', true, warning);
    }

    return new StandardPrintResult('ok', false);

  } catch (error) {
    if (error instanceof StarIO10InUseError) {
      await (new Promise(r => setTimeout(r, 45000)));
      throw new StandardPrintError('in_use_error', error?.message, { location: 'final_catch' });
    } else if (error instanceof StandardPrintError) {
      throw error;
    } else {
      throw new StandardPrintError('unknown', error?.status?.()?.toString?.(), { location: 'final_catch' });
    }

  } finally {
    await printer.close();
    await printer.dispose();
  }
```

**Comment by gare-bear** [comment]
@emiller can you post your device specs? 
Need the OS, Software Version and Model Number

**Comment by user512** [comment]
Hi @gare-bear, I work with @emiller and here is the device spec.
_iPad Air (4th generation), Model:J308AP, running iOS 14.4.2_

And here is our Sentry public link: https://sentry.io/share/issue/7a0894d60d3a42f78d1f240f70da0cc4/

**Comment by gare-bear** [comment]
@user512 Hi Tom, thanks a bunch for sharing this. I'll ask @bandit-ibayashi to review this when he has a chance.

**Comment by bandit-ibayashi** [comment]
@emiller @user512 Thank you for your report and giving us the code snippet.
I made a confirmation app based on this code and am now confirming the reproduction.

I haven't been able to reproduce the crash yet, but could you tell us how often it happens?

**Comment by user512** [comment]
I wasn't sure what's a good way to measure that, but here's a screenshot showing the frequency of the crash.
<img width="1535" alt="Screen Shot 2021-09-20 at 9 25 29 PM" src="https://user-images.githubusercontent.com/8117421/134111446-f3fa124f-f0ea-49c0-9961-4728d4471767.png">

I have personally "reproduce" this bug by quitting the app (swipe up when in app drawer) and upon quitting the app, Sentry informed me the app has crashed.

**Comment by jamiesunderland** [comment]
It's exactly the same issue actually. However, the compile errors people are complaining about in https://github.com/star-micronics/react-native-star-io10/issues/7 are actually good and the desired result. People are getting around them by excluding architectures and messing with framework paths. It's great until you end up with runtime access errors like this one. It's easy to forget compilers are our friends and not our foes.

Though side note on #7, turning flipper on in a library makes it unusable in other libraries since it has to be statically compiled and this library is using modular headers (for the most part).  I'll comment in that thread.

It's not a bug to support simulators and x86. The problem is that as is, the pod is not setup correctly to cross compile archs and is forcing people to cheat the compiler and then get hard to debug crashes at runtime.

I can post a PR as a temporary patch if you think that would be of benefit to you guys

**Comment by gare-bear** [comment]
I think that would be great. I'm sure I can convince @bandit-ibayashi to put the temporary fix in another branch until we can build a more complete solution.

**Comment by joarwilk** [comment]
Sorry to bother but any progress on this task? @gare-bear

**Comment by gare-bear** [comment]
@joarwilk We're hoping to push an update some time this month. 
If you have an urgent need for an update, you should contact your local Star subsidiary and ask about the possibility of a beta release.

**Comment by gare-bear** [comment]
If you're still running into this issue and are willing to test a beta release, please contact the [integration team](https://www.starmicronics.com/pages/contact-us-product-integration) to request beta access.

**Comment by bandit-ibayashi** [comment]
We have updated our React Native SDK including @jamiesunderland suggestion.
We appreciate your patience and great contribution!

**Comment by gare-bear** [comment]
@emiller Since you opened this issue, can you update to [1.1.0](https://github.com/star-micronics/react-native-star-io10/tree/V1.1.0) to verify the issue is fixed?

**Comment by yachaka** [comment]
I had this issue anytime I would close the app.
It got resolved with 1.1.0 👍
